### PR TITLE
NICTIZ-18056: created Symptom profile + updated extensions

### PR DESCRIPTION
**Pull request by njnikodym**
There is one outstanding item i.e. the addition of SymptomStatusDate. This element is not part of the transaction. However, the definition of SymptomSeverity includes the SymptomStatusDate. This is currently a point of discussion within the team.

### CODE REVIEW

File: Profiles/cio-Symptom.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231214" />
+        <map value="cio-dataelement-20-901" />
+        <comment value="SymptomName" />
+      </mapping>
+    </element>
+    <element id="Observation.code.text">
+      <path value="Observation.code.text" />
+      <short value="FurtherSpecificationSymptomName" />
+      <definition value="More detailed specification of the symptomName that has been selected from the code list, because that list does not (yet) contain the required level of detail." />
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:102]: 'symptomName' is very ugly; please let that be changed to 'name of the symptom' (compare with FurtherSpecificationDiagnosisName in the HypersensitivityIntolerance building block) in the data set itself, and then update this description accordingly.

File: Profiles/cio-Symptom.xml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231214" />
+        <map value="cio-dataelement-20-902" />
+        <comment value="FurtherSpecificationSymptomName" />
+      </mapping>
+    </element>
+    <element id="Observation.subject">
+      <path value="Observation.subject" />
+      <min value="1" />
+      <type>
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:113]: I'd remove the references to Group, Device and Location, since those won't be used within the CiO use cases (compare with e.g. cio-Condition and cio-SurveillanceDecision). In nl-core profiles you would not remove such references, but in an information standard profile it makes sense to restrict it more.

File: Profiles/cio-Symptom.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  <copyright value="Copyright and related rights waived via CC0, https://creativecommons.org/publicdomain/zero/1.0/. This does not apply to information from third parties, for example a medical terminology system. The implementer alone is responsible for identifying and obtaining any necessary licenses or authorizations to utilize third party IP in connection with the specification or otherwise." />
+  <fhirVersion value="4.0.1" />
+  <mapping>
+    <identity value="cio-dataset-200-beta2-20231214" />
+    <uri value="https://decor.nictiz.nl/pub/cio/cio-html-20231214T100742/ds-2.16.840.1.113883.2.4.3.11.60.26.1.2-2020-04-02T000000.html" />
+    <name value="ART-DECOR Dataset Contraindications and hypersensitivities 2.0.0-beta.2 20231214" />
+  </mapping>
+  <kind value="resource" />
+  <abstract value="false" />
+  <type value="Observation" />
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:29]: I'm still contemplating whether Observation is the right resource to map a Symptom on, especially due to concepts such as DurationWhenPresent. On the other hand, the FHIR specification states the following:

_This resource is not typically used to record information about subjective and objective information that might lead to the recording of a Condition resource. Such signs and symptoms are typically captured using the [Observation](http://hl7.org/fhir/R4/observation.html) resource; although in some cases a persistent symptom, e.g. fever, headache may be captured as a condition before a definitive diagnosis can be discerned by a clinician._

So in seems that Observation might be the correct resource, since there is always an accompanying Condition resource (corresponding to the Condition building block).

Short version: Observation seems the way to go!

File: Profiles/cio-Symptom.xml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Patient" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Group" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Device" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Location" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-Patient" />
+      </type>
+    </element>
+    <element id="Observation.interpretation">
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:122]: I don't think this mapping is correct. The definition of `Observation.interpretation` is:

_A categorical assessment of an observation value. For example, high, low, normal._

In this profile there currently is no value (the value could be something like 'Present' or 'Absent', see e.g. http://hl7.org/fhir/R4/observation-example-vomiting.xml.html), so it doesn't make sense to make an interpretation on that. Perhaps `Observation.component' is better suited to map the SymptomSeverity concept on.

File: Profiles/cio-Symptom.xml [hunk 5]
```diff
@@ -0,0 +1,153 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:1]: It is really curious that no date/time concept from the data set is present (yet) in the transactions, and hence in this profile. It seems that a Symptom without time stamp would give a lot less information, since you wouldn't know when it occurred, or since when it occurs. According to the ticket, this is still an ongoing discussion?

File: Profiles/cio-Symptom.xml (hunk 4 above)
Comment (LuudSlagter) [reply Profiles/cio-Symptom.xml:122]: Could you check if other zibs have similar Severity concepts, and how these are mapped in FHIR?

File: Profiles/cio-Symptom.xml [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  <publisher value="Nictiz" />
+  <contact>
+    <name value="Nictiz" />
+    <telecom>
+      <system value="url" />
+      <value value="https://www.nictiz.nl" />
+      <use value="work" />
+    </telecom>
+  </contact>
+  <description value="A symptom is a phenomenon with which a Observation in the patient presents itself. This also includes a complaint or a finding, but not additional research, such as laboratory tests, function tests and imaging studies." />
```
[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:18]: Fixed it.

File: Profiles/cio-Symptom.xml [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  <differential>
+    <element id="Observation">
+      <path value="Observation" />
+      <short value="Symptom" />
+      <definition value="A phenomenon with which a condition in the patient presents itself." />
+      <comment value="This also includes a complaint or a finding, but not additional research, such as laboratory tests, function tests and imaging studies." />
+      <alias value="Symptoom" />
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231214" />
+        <map value="cio-dataset-200-beta2-890" />
```
[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:41]: Fixed it.

File: Profiles/cio-Symptom.xml [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+  <abstract value="false" />
+  <type value="Observation" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Observation" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Observation">
+      <path value="Observation" />
+      <short value="Symptom" />
+      <definition value="A phenomenon with which a condition in the patient presents itself." />
+      <comment value="This also includes a complaint or a finding, but not additional research, such as laboratory tests, function tests and imaging studies." />
```
[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:37]: Removed defintition and comment.

File: Profiles/cio-Symptom.xml (hunk 2 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:113]: Makes sense. Removed the references.

File: Profiles/cio-Symptom.xml (hunk 5 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:1]: Yes, this is an ongoing discussion. I would expect the StatusDate to be included in the transaction to capture this information. Specifically because RegistratieDatumTijd is not fit to capture this information.

File: Profiles/cio-Symptom.xml (hunk 4 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:122]: Ah right, .interpretation is not a correct mapping then. Unfortunately there are no Severity concepts that map to something in an Observation that I could take inspiration from. Most of the time there is a dedicated `.severity` element. Such as in Condition. However, observation.component is not either imo. Definition of `Observation.component` is:

_Some observations have multiple component observations. These component observations are expressed as separate code value pairs that share the same attributes. Examples include systolic and diastolic component observations for blood pressure measurement and multiple component observations for genetics observations._

Maybe an extension? Or using a Condition for Symptom in the end?

File: Profiles/cio-Symptom.xml (hunk 1 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:102]: Raised this with the IA. Will adjust once the data set is adjusted.

File: Profiles/cio-Symptom.xml (hunk 3 above)
Comment (LuudSlagter) [reply Profiles/cio-Symptom.xml:29]: If the Symptom zib would be mapped to a Condition profile in nl-core, it's better to already implement it that way in CiO as well (even though elements such as Beloop are not that relevant for CiO (yet)), since that reduces rework in the future and is also a better 'message' to implementers.

Comment (njnikodym) [reply Profiles/cio-Symptom.xml:29]: I'm not certain about the nl-core profile being mapped to Condition. I was not able to find any analysis done by the nl-core team

File: Profiles/cio-Symptom.xml (hunk 4 above)
Comment (LuudSlagter) [reply Profiles/cio-Symptom.xml:122]: Additional food for thought; for the nl-core profiles we had a discussion on `.component` versus `.hasMember`, and there we use the following guideline:

Only use `Observation.component` for any supporting functional zib element that cannot reasonably be interpreted and used outside the scope of the Observation it is a component of. Otherwise, a separate profile on Observation is created for this functional element, which is referenced in `Observation.hasMember` of the 'main' profile. Looking at the DefinitionCode associated with a concept can help identifying this. If the code is a descendant of SNOMED 404684003 (Clinical finding) this could indicate a separate Observation.

For instance, BodyWeight.Clothing or O2Saturation.ExtraOxygenAdministration are mapped on `.component`. These concepts give additional information about the measurement of the main value and can therefore not be considered as separate Observations. Since the severity doesn't make much sense as a separate Observation as well, I still think it'd be justified to map it on `.component`. For Symptom there is not really a `.value`, however the 'measurement' in this case is the acknowledgement of there being a Symptom (which is indicated by `.code`). Again, it would be nice to discuss this with Pieter to see what the latest insights are from an nl-core point of view.

File: Profiles/cio-Symptom.xml [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-MedicationHypersensitivityIdentifier" />
+      </type>
+      <isModifier value="false" />
+    </element>
+    <element id="Observation.extension:medicationHypersensitivityIdentifier.url">
+      <path value="Observation.extension.url" />
+      <fixedUri value="http://nictiz.nl/fhir/StructureDefinition/ext-MedicationHypersensitivityIdentifier" />
+    </element>
+    <element id="Observation.extension:relationCondition">
```
[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:62]: @LuudSlagter  I haven't thought of mapping RelationCondition via `Condition.evidence.detail` but I do support that idea for Symptom. As the zib page describes: _De Aandoening of gesteldheid presenteert zich met symptomen_. So linking the Symptom through the `.evidence` element seems logical. 

However, I'm not a fan of mapping Reaction and Hypersensitivity through `.evidence.detail`. I don't think it is correct to say that Condition has become a more central element. A SurveillanceDecision is primarily linked to a Reaction or a Hypersensitivity via DecisionBasis. A Condition used more as context for the Reaction and Hypersensitivity and not the other way around (the Reaction and Hypersensitivity as "evidence" for the Condition. 

Based on this, I would opt for a hybrid situation.

File: Profiles/cio-Symptom.xml (hunk 4 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:122]: Decision to map element to Observation.component. Based on : http://hl7.org/fhir/R5/observation.html#gr-comp

File: Profiles/cio-Symptom.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:62]: @LuudSlagter I have analyzed this a bit further and I don't think the current Symptom zib en dataset structure is fit for the mapping of RelationCondition through `Condition.evidence.detail`. The name of the dataset element is "Relation**Condition**" and the definition is "_The condition of which the symptom is a manifestation_" which does not make sense to map inside of a Condition profile. Changing the name and description would be confusing and does not make sense. I wanted to suggest to IA's to move the relation between Condition and Symptom to go from Condition to Symptom instead of Symptom to Condition. However, the zib page states that there is a link from Symptom to Condition and not the other way around. I assume we cannot simply change the direction without any consequences. 
I will discuss this point with IA's.

[the remark this answers is not in the document]
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:62]: @LuudSlagter okay so I thought about this a little bit and checked it with Alexander. In conclusion I chose to omit the mapping of RelationCondition and added a relation between Condition and Symptom through `.evidence.detail`. I will also add some supporting comments to Symptom and Condition profile.

File: Profiles/cio-Symptom.xml (hunk 1 above)
Comment (njnikodym) [reply Profiles/cio-Symptom.xml:102]: IA's do not want to change the description as it is the literal description used in the zib. Conclusion: I will not change it.

File: Profiles/cio-Symptom.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (LuudSlagter) [reply Profiles/cio-Symptom.xml]: See https://informatiestandaarden.nictiz.nl/wiki/FHIR:V1.0_FHIR_Profiling_Guidelines_R4#References_that_are_reversed_in_FHIR for guidance on how to model such a concept.

File: Profiles/cio-Symptom.xml [hunk 10]
```diff
@@ -33,39 +33,20 @@
...
       <short value="Symptom" />
-      <definition value="A phenomenon with which a condition in the patient presents itself." />
-      <comment value="This also includes a complaint or a finding, but not additional research, such as laboratory tests, function tests and imaging studies." />
+      <comment value="This profile is always used together with an instance conforming to [cio-Condition](http://nictiz.nl/fhir/StructureDefinition/cio-Condition) and [cio-Reaction](http://nictiz.nl/fhir/StructureDefinition/cio-Reaction). &#xD;&#xA;&#xD;&#xA;Note that the concept RelationCondition from the Symptom building block is not mapped within this profile. Instead, the relationship between Symptom and Condition is established through the `Condition.evidence.detail` element. Please refer to the [cio-Condition](http://nictiz.nl/fhir/StructureDefinition/cio-Condition) profile for more guidance." />
       <alias value="Symptoom" />
       <mapping>
-        <identity value="cio-dataset-200-beta2-20231214" />
-        <map value="cio-dataset-200-beta2-890" />
+        <identity value="cio-dataset-20-20231214" />
+        <map value="cio-dataset-20-890" />
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:40]: This should be 'cio-data**element**-...' (and similarly for all other concepts).

**Review by LuudSlagter — CHANGES_REQUESTED** [review]

File: Profiles/cio-Symptom.xml [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+        <identity value="cio-dataset-20-20231214" />
+        <map value="cio-dataelement-20-780" />
+        <comment value="IdentificationNumber" />
+      </mapping>
+    </element>
+    <element id="Observation.code">
+      <path value="Observation.code" />
+      <short value="SymptomName" />
+      <definition value="Name and code of the symptom via selection from a code list. Represents the symptom which a condition in the patient presents itself." />
+      <alias value="SymptoomNaame" />
```
Comment (LuudSlagter) [inline Profiles/cio-Symptom.xml:68]: SymptoomNaam

**Review by LuudSlagter — CHANGES_REQUESTED** [review]

File: Profiles/cio-Symptom.xml (hunk 1 above)
Comment (LuudSlagter) [reply Profiles/cio-Symptom.xml:83]: Has this typo already been communicated to the zib centre, so it can be fixed in a later version?

File: Profiles/cio-Symptom.xml (hunk 11 above)
Comment (alayabenbelkacem) [reply Profiles/cio-Symptom.xml:68]: fixed

File: Profiles/cio-Symptom.xml (hunk 10 above)
Comment (alayabenbelkacem) [reply Profiles/cio-Symptom.xml:40]: fixed

**Review by LuudSlagter — APPROVED** [review]
Additional review will be done in https://github.com/Nictiz/Nictiz-R4-cio/pull/8.
